src/app.rs

```diff
@@ -245,6 +276,36 @@ impl App {
         Ok(InclusionProofResponse(proof))
     }
 
+    /// # Errors
+    ///
+    /// Will return `Err` if the provided proof is invalid.
+    #[instrument(level = "debug", skip_all)]
+    pub async fn verify_proof(
+        &self,
+        request: &VerifyProofRequest,
+    ) -> Result<VerifyProofResponse, ServerError> {
+        let Some(root_timestamp) = self.database.get_root_timestamp(&request.root).await? else {
+            return Ok(VerifyProofResponse::InvalidRoot)
+        };
+
+        if root_timestamp + self.root_validity_timeout < Utc::now() {
```

Comment:
This is too simplistic. Suppose inserts stopped for 12 hrs. Then the latest root will be deemed invalid. We've already changed this in the contract such that:
1. Latest root is always valid.
2. Root expiration starts counting when the root is superseded, not when it's created.

Essentially, if we stop inserts for 12 hrs, then this long-lived root should not become invalid as soon as next insert comes, but rather it should remain valid for a set period.

Reply:
Fixed.